### ISSUE TITLE
kernel panic in lxpr_read_pid_env

### ISSUE THREAD

**Issue by rzezeski** [joyent/illumos-joyent#94]
A user was able to panic the entire node while compiling git in an lx-brand zone.

```
panic[cpu0]/thread=fffff04322cf53c0:
recursive mutex_enter, lp=fffff042e0085140 owner=fffff04322cf53c0 thread=fffff04322cf53c0


fffff001e8364260 unix:mutex_panic+58 ()
fffff001e83642d0 unix:mutex_vector_enter+2b7 ()
fffff001e8364340 genunix:anon_resvmem+68 ()
fffff001e8364460 genunix:segvn_faultpage+216 ()
fffff001e8364630 genunix:segvn_fault+8e6 ()
fffff001e8364760 genunix:as_fault+37a ()
fffff001e83647f0 unix:pagefault+96 ()
fffff001e8364900 unix:trap+d88 ()
fffff001e8364910 unix:cmntrap+e6 ()
fffff001e8364a20 unix:bcopy_altentry+55a ()
fffff001e8364a90 genunix:uiomove+f8 ()
fffff001e8364ad0 lx_proc:lxpr_uiobuf_flush+84 ()
fffff001e8364b20 lx_proc:lxpr_uiobuf_write+98 ()
fffff001e8364b70 lx_proc:lxpr_read_pid_env+71 ()
fffff001e8364c20 lx_proc:lxpr_read+63 ()
fffff001e8364cc0 genunix:fop_read+8b ()
fffff001e8364e20 genunix:read+2a7 ()
fffff001e8364e70 lx_brand:lx_read+79 ()
fffff001e8364ef0 lx_brand:lx_syscall_enter+16f ()
fffff001e8364f10 unix:brand_sys_syscall+1e7 ()
```

```
> fffff043555c2068::walk thread | ::print kthread_t t_procp | ::print proc_t ! egrep 'zone|psargs'
    p_zone = 0xfffff042eab80d00
        u_psargs = [ "/usr/bin/ld --eh-frame-hdr --build-id -m elf_x86_64 --hash-style=gnu -dynamic-l" ]
> fffff042eab80d00::print zone_t zone_name zone_brand
zone_name = 0xfffff0447f978cc0 "f3a66ca2-ab61-4c38-9a0a-25590113443b"
zone_brand = lx_brand
```

The problem is that the `lxpr_read_pid_env()` is holding the pid lock when it calls uiomove, which can cause an anon page to get mapped in, which also grabs the pid lock.

Webrev incoming.

**Comment by rzezeski** [comment]
Webrev: http://zinascii.com/pub/illumos/joyent/fix-94/

**Comment by pfmooney** [comment]
Merged in 5f61927a04d66e0d1f42a7c0f1572de01debacae.
Thanks
